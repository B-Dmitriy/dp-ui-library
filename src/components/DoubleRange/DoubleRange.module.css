.doubleRange {
    padding: 20px;
}