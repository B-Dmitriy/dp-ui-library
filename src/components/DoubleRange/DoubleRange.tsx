import React from 'react';
import s from './DoubleRange.module.css';
import {Range} from 'rc-slider';
import 'rc-slider/assets/index.css';

interface IProps {
    min: number
    max: number
    step?: number
    value: number[]
    defaultValue?: number[]
    onChange: (value: number[]) => void
}

export const DoubleRange: React.FC<IProps> = props => {

    const {
        min,
        max,
        step = 1,
        value,
        defaultValue,
        onChange
    } = props;

    return (
        <div className={s.doubleRange} data-testid={'doubleRange_wrapper'}>
            <Range
                min={min}
                max={max}
                step={step}
                value={value}
                defaultValue={defaultValue}
                onChange={onChange}
                railStyle={{
                    backgroundColor: 'lightgrey',
                    height: '10px'
                }}
                trackStyle={[{
                    backgroundColor: '#259cd3',
                    boxShadow: '0 15px 20px rgb(10, 136, 196, .4)',
                    height: '10px'
                }]}
                handleStyle={[{
                    position: 'absolute',
                    top: '-1px',
                    width: '16px',
                    height: '32px',
                    borderRadius: '10px',
                    border: '2px solid #259cd3',
                    boxShadow: '0 15px 20px rgb(10, 136, 196, .4)',
                }, {
                    position: 'absolute',
                    top: '-1px',
                    width: '16px',
                    height: '32px',
                    borderRadius: '10px',
                    border: '2px solid #259cd3',
                    boxShadow: '0 15px 20px rgb(10, 136, 196, .2)',
                }]}
            />
        </div>
    );
};