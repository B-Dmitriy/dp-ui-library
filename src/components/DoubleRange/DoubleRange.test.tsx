import {render, screen} from '@testing-library/react';
import {DoubleRange} from './DoubleRange';

const onChange = jest.fn();

describe('DoubleRange test`s', () => {

    it('should be render', () => {
        render(<DoubleRange
            min={1}
            max={10}
            value={[3, 4]}
            onChange={onChange}
        />);

        expect(screen.getByTestId('doubleRange_wrapper')).toBeInTheDocument();
    });

    it('checkbox snapshot', () => {
        const view = render(<DoubleRange
            min={1}
            max={10}
            value={[3, 4]}
            onChange={onChange}
        />);

        expect(view).toMatchSnapshot();
    });
});