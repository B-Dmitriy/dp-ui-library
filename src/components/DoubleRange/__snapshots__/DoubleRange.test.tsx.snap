// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoubleRange test\`s checkbox snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="doubleRange"
        data-testid="doubleRange_wrapper"
      >
        <div
          class="rc-slider"
        >
          <div
            class="rc-slider-rail"
            style="background-color: lightgrey; height: 10px;"
          />
          <div
            class="rc-slider-track rc-slider-track-1"
            style="background-color: rgb(37, 156, 211); box-shadow: 0 15px 20px rgb(10, 136, 196, .4); height: 10px; left: 22.22222222222222%; width: 11.111111111111107%;"
          />
          <div
            class="rc-slider-step"
          />
          <div
            aria-disabled="false"
            aria-valuemax="10"
            aria-valuemin="1"
            aria-valuenow="3"
            class="rc-slider-handle rc-slider-handle-1"
            role="slider"
            style="position: absolute; top: -1px; width: 16px; height: 32px; border-radius: 10px; border: 2px solid #259cd3; box-shadow: 0 15px 20px rgb(10, 136, 196, .4); left: 22.22222222222222%; transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            aria-disabled="false"
            aria-valuemax="10"
            aria-valuemin="1"
            aria-valuenow="4"
            class="rc-slider-handle rc-slider-handle-2"
            role="slider"
            style="position: absolute; top: -1px; width: 16px; height: 32px; border-radius: 10px; border: 2px solid #259cd3; box-shadow: 0 15px 20px rgb(10, 136, 196, .2); left: 33.33333333333333%; transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            class="rc-slider-mark"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="doubleRange"
      data-testid="doubleRange_wrapper"
    >
      <div
        class="rc-slider"
      >
        <div
          class="rc-slider-rail"
          style="background-color: lightgrey; height: 10px;"
        />
        <div
          class="rc-slider-track rc-slider-track-1"
          style="background-color: rgb(37, 156, 211); box-shadow: 0 15px 20px rgb(10, 136, 196, .4); height: 10px; left: 22.22222222222222%; width: 11.111111111111107%;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="10"
          aria-valuemin="1"
          aria-valuenow="3"
          class="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style="position: absolute; top: -1px; width: 16px; height: 32px; border-radius: 10px; border: 2px solid #259cd3; box-shadow: 0 15px 20px rgb(10, 136, 196, .4); left: 22.22222222222222%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          aria-disabled="false"
          aria-valuemax="10"
          aria-valuemin="1"
          aria-valuenow="4"
          class="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style="position: absolute; top: -1px; width: 16px; height: 32px; border-radius: 10px; border: 2px solid #259cd3; box-shadow: 0 15px 20px rgb(10, 136, 196, .2); left: 33.33333333333333%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
